src/hotspot/cpu/x86/nativeInst_x86.cpp

```diff
@@ -510,12 +510,28 @@ void NativeJump::check_verified_entry_alignment(address entry, address verified_
 //
 void NativeJump::patch_verified_entry(address entry, address verified_entry, address dest) {
   // complete jump instruction (to be inserted) is in code_buffer;
+#ifdef AMD64
+  unsigned char code_buffer[8];
```

Comment:
I suggest using a union.

Reply:
@dean-long  
Complete decomposition to union, like one below, looks really nice, but requires gcc-specific attribute, that I would like to avoid. 
```
  union {
    jlong cb_long;
    struct {
      char code;
      int32_t disp;
    } __attribute__((packed)) instr;
  } u;
  ```
  Do you prefer  
  
  ```
  union {
    jlong cb_long;
    unsigned char code_buffer[8];
 } u;
 ``` 
 over the cast that is similar to one in 32bit version?

Reply:
@vnkozlov 

CXX optimizes that code to a few of register operations, and optimize out local variable _code_buffer_,  so we need not to care about its alignment. 

```   mov    (%r12),%rdx
      mov    %ebx,%eax
      shl    $0x8,%rax
      mov    $0xe9,%dl
      and    %rcx,%rdx
      mov    %rax,(%r12)
```

Reply:
> ```
> union {
>   jlong cb_long;
>   unsigned char code_buffer[8];
> } u;
> ```
> 

This version is fine.  It removes any question about the alignment of cb_long at the same time.

Reply:
> This version is fine. It removes any question about the alignment of cb_long at the same time.

I agree.

Reply:
done